src/lib.rs

```diff
@@ -538,79 +556,94 @@ where
                 );
             }
 
-            base_address = match entry.attr_value(gimli::DW_AT_low_pc) {
-                Ok(Some(gimli::AttributeValue::Addr(addr))) => addr,
-                Err(e) => {
-                    return Err(Error::from(ErrorKind::Gimli(e)))
-                        .chain_err(|| "invalid low_pc attribute")
-                }
-                _ => match entry.attr_value(gimli::DW_AT_entry_pc) {
-                    Ok(Some(gimli::AttributeValue::Addr(addr))) => addr,
-                    Err(e) => {
-                        return Err(Error::from(ErrorKind::Gimli(e)))
-                            .chain_err(|| "invalid entry_pc attribute")
+            let mut attrs = entry.attrs();
+            while let Some(attr) = attrs
+                .next()
+                .map_err(|e| Error::from(ErrorKind::Gimli(e)))
+                .chain_err(|| "invalid unit attribute")?
+            {
+                match attr.name() {
```

Comment:
This change means that we no longer error out if we find that a compilation unit is missing a unit statement list, compilation unit name, compilation unit directory, etc. That seems unfortunate?

Reply:
This behaviour hasn't been changed; we already didn't return an error if they were missing. The only change is that if the attribute does exist and we failed to parse it, then the error returned is now less specific.

Reply:
Actually, the error wasn't all the specific before anyway, because the error could have occurred on an attribute leading up to the one we were looking for.